Update Auto-Updator

GamingLiamStudios opened the pull request:
The auto-update feature now fetches the newest build from circle-ci.
You can also change the branch which said builds will come from.

And of course, OWO translation for it all.
oh, and vscode doing a bunch of formatting, as usual.

coredev-uk commented:
Please revert the formatting if possible, its very hard to track the changed that you've made and the formatting that you've done will most likely get reverted.

coredev-uk commented:
I'll leave this as a draft. Change it back when you are ready.

GamingLiamStudios commented:
I've gone through and fixed a lot of the formatting that vscode fucked up. Still quite a bit of changes tho, but it's 2.10am, they look nice and I just want to sleep.

GamingLiamStudios commented:
General rundown of what I did do:
- Added new dropdown menu under 'Update Cider' button to choose which branch to update from
- Added circle-ci API call to get most recent build from given branch
- Replaced `term.updateCider` to `settings.option.experimental.updateCider`
- OWO translation of all the new settings
- Some formatting(blame vscode. except for i18n files, that was me.)

quacksire commented on src/renderer/views/pages/settings.ejs:
```diff
@@ -655,14 +655,30 @@
+                        {{$root.getLz('settings.option.experimental.updateCider')}} 
                     </div>
                     <div class="md-option-segment md-option-segment_auto">
-                        <button class="md-btn" @click="app.checkForUpdate()">
```
keep this. for vue

GamingLiamStudios commented:
Shit. Don't merge, somehow broke cider. Probably with that upstream merge.

GamingLiamStudios commented:
And fixed. was a stray bracket.

GamingLiamStudios commented:
Also, before we do merge this, I want to propose a new feature for the Auto-Updater. A popup saying the status of the update, such as `Cider is already up to date` or `Your platform does not support Auto-Updates`
I'll look into how this might work, as it would make Updating feel a bit more responsive.
